data-prepper-core/src/main/java/org/opensearch/dataprepper/parser/model/DataPrepperConfiguration.java

```diff
@@ -61,6 +66,7 @@ public DataPrepperConfiguration(
         setMetricTags(metricTags);
         setServerPort(serverPort);
         this.peerForwarderConfiguration = peerForwarderConfiguration;
+        this.processorShutdownTimeout = processorShutdownTimeout != null ? processorShutdownTimeout : DEFAULT_SHUTDOWN_DURATION;
```

Comment:
Related to some of my questions below about negative numbers, we may wish to verify that the Duration is not negative.

Maybe `if(processorShutdownTimeout.isNegative()) throw new SomeException()`.

Reply:
That's a good idea, will add